### ISSUE TITLE
[a11y]: eds-tree-view: The node labels are cut off and illegible on narrow trees.

### ISSUE THREAD

**Issue by rsrduarteEG** [carbon-design-system/carbon#21541]
### Package

@carbon/web-components

### Package version

2.48.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

When using the eds-tree-view component with a small width and nodes containing long labels, the labels are cut off, making it difficult or impossible for users—especially those relying on assistive technologies—to read the full content. This impacts accessibility and usability, as important information in the node labels may be hidden or unreadable.

### Reproduction/example

https://stackblitz.com/edit/github-xsekvhst?file=index.html

### Steps to reproduce

**Steps to Reproduce:**
1. Check the eds-tree-view example in the stackblitz
2. Observe that the node labels are visually cut off, with no way to view the full label (e.g., no text wrapping, ellipsis, or accessible tooltip).

**Expected Result:**
The full label text should be accessible to all users, either by wrapping, showing an ellipsis with a tooltip, or providing another accessible method to read the entire label (e.g., screen readers can access the full text).

**Actual Result:**
The label text is visually cut off, with no way to view the full label. Users—including those using assistive technologies—cannot read or access the complete label, making navigation and understanding difficult.

<img width="137" height="200" alt="Image" src="https://github.com/user-attachments/assets/e9a55772-aba4-49bc-8e4f-bdad85fa4427" />

<img width="134" height="196" alt="Image" src="https://github.com/user-attachments/assets/0b242aca-a0e8-46ba-b6b7-593b7f1d348a" />

In the React version of the tree view, this issue is resolved by cutting the label, adding ellipsis, and adding a tooltip with the complete label.

<img width="389" height="232" alt="Image" src="https://github.com/user-attachments/assets/294515a6-7f11-4d09-81ce-672eb577bc9c" />

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
